factory/README.md

```diff
@@ -0,0 +1,201 @@
+
+### NODE_VERSION
+
+The version of Node to install in the docker container. If not specified, the default version of Node (defined [here](./.env)) is installed. Node is required. The exact version must be used, no wildcards or shorthands are supported.
+
+Example: `NODE_VERSION='16.18.1'`
+
+[Node Versions](https://nodejs.org/en/download/releases)
+
+### YARN_VERSION
+
+The version of yarn to install (via npm). If not specified, Yarn is not installed.
+
+Example: `YARN_VERSION='1.22.19'`
+
+[Yarn versions](https://www.npmjs.com/package/yarn)
+
+### CYPRESS_VERSION
+
+The version of Cypress to install (via npm). If not specified, Cypress is not installed.
+
+Example: `CYPRESS_VERSION='12.1.0'`
+
+[Cypress versions](https://www.npmjs.com/package/cypress)
+
+### CHROME_VERSION
+
+The version of Chrome to install. If not specified, Chrome is not installed. The exact version must be used, no wildcards or shorthands are supported.
+
+Example: `CHROME_VERSION='107.0.5304.121-1'`
+
+[Chrome versions](https://www.ubuntuupdates.org/package/google_chrome/stable/main/base/google-chrome-stable)
+
+### FIREFOX_VERSION
+
+The version of Firefox to install. If not specified, Firefox is not installed. The exact version must be used, no wildcards or shorthands are supported.
+
+Example: `FIREFOX_VERSION='107.0'`
+
+[Firefox versions](https://download-installer.cdn.mozilla.net/pub/firefox/releases/)
+
+### EDGE_VERSION
+
+The version of Edge to install. If not specified, Edge is not installed. The exact version must be used, no wildcards or shorthands are supported.
+
+Example: `EDGE_VERSION='100.0.1185.29-1'`
+
+[Edge versions](https://packages.microsoft.com/repos/edge/pool/main/m/microsoft-edge-stable/)
+
+## Usage
+
+The cypress docker factory works by relying on the [`ONBUILD`](https://docs.docker.com/engine/reference/builder/#onbuild) docker instruction to run commands at the container's build time. Docker `args` can be specified in a number of ways, some of which are demonstrated below. For each of these examples we are building the equivalent of the `cypress/browsers` docker image. In each instance since the example is only testing the chrome version, the examples could just install chrome by itself if the other browsers were not used.
+
+### In the Dockerfile
+
+Args can be defined directly in the Dockerfile to specify variables that are available when the container is built.
+
+Dockerfile
+
+```dockerfile
+# Args are defined in the Dockerfile before the FROM command.
+# Using these args will cause an image to be created with node (default version is 16.18.1), chrome, firefox and edge.
+ARG CHROME_VERSION='107.0.5304.121-1'
+ARG EDGE_VERSION='100.0.1185.29-1'
+ARG FIREFOX_VERSION='107.0'
+
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+Then, is the same directory as the Dockerfile, run the following commands to build the docker container and run Cypress against the chrome browser.
```

Comment:
Should this read _in_ the same directory?

Reply:
yes
```suggestion
Then, in the same directory as the Dockerfile, run the following commands to build the docker container and run Cypress against the chrome browser.
```